packages/replay/src/replay.ts

```diff
@@ -824,10 +824,19 @@ export class ReplayContainer implements ReplayContainerInterface {
         timestamp: new Date().getTime(),
       });
     } catch (err) {
+      this._handleException(err);
+
       if (err instanceof RateLimitError) {
         this._handleRateLimit(err.rateLimits);
+        return;
       }
-      this._handleException(err);
+
+      const client = getCurrentHub().getClient();
+      client?.recordDroppedEvent('retry_failed', 'replay_event');
```

Comment:
Aren't we trying to not use null coalescing?

Reply:
Good point! This will be in another PR anyhow now. Also, actually I'll make a PR updating all current usages of optional chaining in replay.

Reply:
https://github.com/getsentry/sentry-javascript/pull/6777